### PR TITLE
Add tooltip support

### PR DESCRIPTION
**Pull request by tonikitoo**
This PR is a set of patches that add support for tooltips in ozone/mus (external window mode).

TBR=msisov

### CODE REVIEW

**Comment by msisov** [comment]
Looks like there is something wrong with tooltips on wayland. 

It crashes gnome shell on my machine: 

Nov  6 08:20:04 msisov-dell kernel: [37456.832740] gnome-shell[13523]: segfault at 30 ip 00007f0becdcc690 sp 00007ffc17e62d00 error 4 in libmutter.so.0.0.0[7f0becd01000+12c000]
Nov  6 08:20:04 msisov-dell org.gnome.Shell.desktop[13523]: (EE)
Nov  6 08:20:04 msisov-dell org.gnome.Shell.desktop[13523]: Fatal server error:
Nov  6 08:20:04 msisov-dell org.gnome.Shell.desktop[13523]: (EE) failed to read Wayland events: Broken pipe
Nov  6 08:20:04 msisov-dell org.gnome.Shell.desktop[13523]: (EE)
Nov  6 08:20:04 msisov-dell xfce4-terminal[13601]: Error reading events from display: Broken pipe

Have you happened to see it?

File: ui/views/mus/desktop_window_tree_host_mus.cc [hunk 1]
```diff
@@ -487,8 +490,13 @@ aura::WindowTreeHost* DesktopWindowTreeHostMus::AsWindowTreeHost() {
 }
 
 void DesktopWindowTreeHostMus::ShowWindowWithState(ui::WindowShowState state) {
+#if defined(OS_LINUX) && defined(USE_OZONE) && !defined(OS_CHROMEOS)
+  // On Ozone/Linux we need to notify the server of any show-state change.
+  window()->SetProperty(aura::client::kShowStateKey, state);
+#else
   if (state == ui::SHOW_STATE_MAXIMIZED || state == ui::SHOW_STATE_FULLSCREEN)
     window()->SetProperty(aura::client::kShowStateKey, state);
```
Comment (msisov) [inline ui/views/mus/desktop_window_tree_host_mus.cc:498]: What would happen if the show property state is changed for chrome os case as well? 
What I want to say is whether it's possible to unify these lines and just have window()->SetProperty() for both OS_LINUX and OS_CHROMEOS.

File: ui/events/platform/x11/x11_event_source_libevent.cc [hunk 2]
```diff
@@ -99,11 +99,10 @@ std::unique_ptr<ui::Event> TranslateXEventToEvent(const XEvent& xev) {
   switch (xev.type) {
     case LeaveNotify:
     case EnterNotify:
-      return std::make_unique<MouseEvent>(ET_MOUSE_MOVED,
-                                          EventLocationFromXEvent(xev),
-                                          EventSystemLocationFromXEvent(xev),
-                                          EventTimeFromXEvent(xev), flags, 0);
-
+      return std::make_unique<MouseEvent>(
+          xev.type == EnterNotify ? ET_MOUSE_MOVED : ET_MOUSE_EXITED,
```
Comment (msisov) [inline ui/events/platform/x11/x11_event_source_libevent.cc:103]: Can you use EventTypeFromFromXEvent instead?

**Comment by msisov** [comment]
I've also noticed that sometimes the string in the tooltip window is stripped and thus a part of the string is hidden.

File: ui/views/mus/desktop_window_tree_host_mus.cc (hunk 1 above)
Comment (tonikitoo) [reply ui/views/mus/desktop_window_tree_host_mus.cc:498]: I think we can try, sir. Lets see what sky@ says.

**Pull request merged**